### PR TITLE
Rewrite DocumentUpdateTest from cppunit to gtest.

### PR DESCRIPTION
**Pull request by geirst**
@vekterli please review
@toregge FYI

### CODE REVIEW

File: document/src/tests/documentupdatetestcase.cpp [hunk 1]
```diff
@@ -550,61 +458,61 @@ void DocumentUpdateTest::testReadSerializedFile()
...
+    EXPECT_EQ(*type, upd.getType());
 
     // Verify assign value update.
     FieldUpdate serField = upd.getUpdates()[1];
-    CPPUNIT_ASSERT_EQUAL(serField.getField().getId(), type->getField("intfield").getId());
+    EXPECT_EQ(serField.getField().getId(), type->getField("intfield").getId());
 
     const ValueUpdate* serValue = &serField[0];
-    CPPUNIT_ASSERT_EQUAL(serValue->getType(), ValueUpdate::Assign);
+    EXPECT_EQ(serValue->getType(), ValueUpdate::Assign);
```
Comment (vekterli) [inline document/src/tests/documentupdatetestcase.cpp]: Consider changing `getType()` checks in this test to `ASSERT_EQ`; blind `static_assert`s are done based on the checks

File: document/src/tests/documentupdatetestcase.cpp [hunk 2]
```diff
@@ -697,18 +594,17 @@ DocumentUpdateTest::testUpdateApplyNoArrayValues()
 
     // Verify that the field was set in the document
     std::unique_ptr<ArrayFieldValue> fval(doc->getAs<ArrayFieldValue>(field));
-    CPPUNIT_ASSERT(fval.get());
-    CPPUNIT_ASSERT_EQUAL((size_t) 0, fval->size());
+    EXPECT_TRUE(fval.get());
```
Comment (vekterli) [inline document/src/tests/documentupdatetestcase.cpp]: Consider `ASSERT_TRUE`, is dereferenced in the following expectation

File: document/src/tests/documentupdatetestcase.cpp [hunk 3]
```diff
@@ -717,8 +613,8 @@ DocumentUpdateTest::testUpdateArrayEmptyParamValue()
 
     // Verify that the field was set in the document.
     std::unique_ptr<ArrayFieldValue> fval1(doc->getAs<ArrayFieldValue>(field));
-    CPPUNIT_ASSERT(fval1.get());
-    CPPUNIT_ASSERT_EQUAL((size_t) 0, fval1->size());
+    EXPECT_TRUE(fval1.get());
```
Comment (vekterli) [inline document/src/tests/documentupdatetestcase.cpp]: Consider `ASSERT_TRUE`, is dereferenced in the following expectation

File: document/src/tests/documentupdatetestcase.cpp [hunk 4]
```diff
@@ -746,8 +641,8 @@ DocumentUpdateTest::testUpdateWeightedSetEmptyParamValue()
 
     // Verify that the field was set in the document.
     auto fval1(doc->getAs<WeightedSetFieldValue>(field));
-    CPPUNIT_ASSERT(fval1.get());
-    CPPUNIT_ASSERT_EQUAL((size_t) 0, fval1->size());
+    EXPECT_TRUE(fval1.get());
```
Comment (vekterli) [inline document/src/tests/documentupdatetestcase.cpp]: Consider `ASSERT_TRUE`, is dereferenced in the following expectation

**Review by vekterli — APPROVED** [review]
Overall looks good, consider changing some expectations to assertions

File: document/src/tests/documentupdatetestcase.cpp (hunk 2 above)
Comment (geirst) [reply document/src/tests/documentupdatetestcase.cpp]: Fixed

File: document/src/tests/documentupdatetestcase.cpp (hunk 3 above)
Comment (geirst) [reply document/src/tests/documentupdatetestcase.cpp]: Fixed

File: document/src/tests/documentupdatetestcase.cpp (hunk 4 above)
Comment (geirst) [reply document/src/tests/documentupdatetestcase.cpp]: Fixed

File: document/src/tests/documentupdatetestcase.cpp (hunk 1 above)
Comment (geirst) [reply document/src/tests/documentupdatetestcase.cpp]: Fixed

**Comment by geirst** [comment]
Thanks, I agree. @vekterli PTAL

**Review by vekterli — APPROVED** [review]
:+1:

**Pull request merged**